client/blocks/term-form-dialog/index.jsx

```diff
@@ -134,11 +135,42 @@ class TermFormDialog extends Component {
 		this.props.bumpStat( 'taxonomy_manager', statLabels.mc );
 		this.props.recordGoogleEvent( 'Taxonomy Manager', statLabels.ga );
 
-		savePromise.then( ( savedTerm ) => {
-			this.setState( { saving: false } );
-			this.props.onSuccess( savedTerm );
-			this.closeDialog();
-		} );
+		savePromise
+			.then( ( savedTerm ) => {
+				this.setState( { saving: false } );
+				this.props.onSuccess( savedTerm );
+				this.closeDialog();
+
+				const successText = isNew
+					? translate( 'Added new %(taxonomyName)s successfully.', {
+							args: { taxonomyName: taxonomy },
+					  } )
+					: translate( 'Updated %(taxonomyName)s successfully.', {
+							args: { taxonomyName: taxonomy },
+					  } );
+
+				this.props.successNotice( successText, {
+					id: 'taxonomy-manager-save',
+					duration: 4000,
+				} );
+			} )
+			.catch( () => {
+				this.setState( { saving: false } );
+				this.closeDialog();
+
+				const errorText = isNew
+					? translate( 'Failed to create new %(taxonomyName)s.', {
```

Comment:
I don't think `catch()` captures the taxonomy term creation failure successfully. I've tried blocking the network request, adding a new term, and still saw the success message, although the request was blocked and the term was not created.

Reply:
I can't reproduce this; it works fine both when I block the request and currently when I don't because of the bug.

<img width="218" alt="Screenshot 2024-03-30 at 21 17 35" src="https://github.com/Automattic/wp-calypso/assets/43215253/f380a379-2b5c-4461-9f25-e350758067e5">
<img width="1302" alt="Screenshot 2024-03-30 at 21 17 41" src="https://github.com/Automattic/wp-calypso/assets/43215253/39ae53fa-a9c0-4308-bb3c-f1806a69e95d">

Reply:
Sorry, I can still reproduce - here's a video - you'll spot that I blocked the network request to the API endpoint for creating a new term:


https://github.com/Automattic/wp-calypso/assets/8436925/50fd5071-49a2-452f-99cd-4b5d0fff45fc

Reply:
Ahh, I see. I'm not sure why it doesn't work for creating a new term - I presume that it works if you block the endpoint which updates a term? 

I can't quite figure out what's different about the API requests here, so I'm a little stumped! However, I have no problems creating a new term normally, and the current bug only seems to affect editing existing terms, so I'm not sure this is too big a deal - but I'll still try to get to the bottom of it.

Reply:
Thanks for taking the time to look into it!

Do you mean you can't reproduce what I've demonstrated? Creating a term with a blocked endpoint and not seeing the error notice?

Anyway, yeah, since the code in question handles both term editing and creation, it makes sense that it covers both scenarios well.

Reply:
Sorry - I dug into this a bit further but didn't have time to report back. The problem here is that `addTerm` intentionally fails silently:

https://github.com/Automattic/wp-calypso/blob/b3bc91bc454f1a8b4e4d2cdbd4d693f210e1dd1c/client/state/terms/actions.js#L48

I've removed that line of code, which means `.catch()` will (should!) now work for both cases. I don't see any problems with that, but I also can't find the reasoning why it was originally included (it's over 8 years old). 

Curious for your thoughts on if you think this change is harmless!

Reply:
Good find! I usually wouldn't recommend concealing errors like this. 

Removing it makes sense at a first glance, but I'll take another look.